### PR TITLE
[Vectorized][Function] Support geolocation functions on vectorized engine

### LINKED ISSUE #8788: [Feature] Support geolocation functions on vectorized engine
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, geolocation functions are not supported on vectorized engine.

I will implement geolocation functions and split those to two PRs which is more easier for reviewing. 

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #8788 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zbtzbtzbt** [comment]
good job!
in addition, please update document in
- docs/en/administrator-guide/vectorized-execution-engine.md 
- docs/zh-CN/administrator-guide/vectorized-execution-engine.md

delete following the statement:
`The geolocation function is not supported, including all functions starting with ST_ in the function. For details, please refer to the section on SQL functions in the official documentation`

**Comment by Gabriel39** [comment]
> good job! in addition, please update document in
> 
> * docs/en/administrator-guide/vectorized-execution-engine.md
> * docs/zh-CN/administrator-guide/vectorized-execution-engine.md
> 
> delete following the statement: `The geolocation function is not supported, including all functions starting with ST_ in the function. For details, please refer to the section on SQL functions in the official documentation`

Thanks @zbtzbtzbt I plan to do this in next PR because I will implement all geolocation functions in next PR which is described in #8788

**Review by zenoyang — COMMENTED** [review]
and it's best to provide function UT

File: be/src/geo/geo_types.cpp [hunk 1]
```diff
@@ -31,6 +34,18 @@
 
 namespace doris {
 
+GeoPoint::GeoPoint() : _point(new S2Point()) {}
```
Comment (HappenLee) [inline be/src/geo/geo_types.cpp:37]: You can go directly  implement these in geo_type.h simple construction destructors

File: be/src/geo/geo_types.h [hunk 2]
```diff
@@ -80,13 +87,13 @@ class GeoPoint : public GeoShape {
     bool decode(const void* data, size_t size) override;
 
 private:
-    S2Point _point;
+    std::unique_ptr<S2Point> _point;
```
Comment (HappenLee) [inline be/src/geo/geo_types.h:90]: why use `unique_ptr` here ?

File: be/src/vec/functions/functions_geo.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    static constexpr auto name = Impl::NAME;
+    static FunctionPtr create() { return std::make_shared<GeoFunction<Impl>>(); }
+    String get_name() const override { return name; }
+    size_t get_number_of_arguments() const override { return Impl::NUM_ARGS; }
+    bool is_variadic() const override { return false; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return make_nullable(std::make_shared<ReturnType>());
+    }
+    bool use_default_implementation_for_nulls() const override { return false; }
```
Comment (HappenLee) [inline be/src/vec/functions/functions_geo.h:39]: both should be true. you can delete the unless code in execute

File: be/src/vec/functions/functions_geo.cpp [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        auto null_type = std::reinterpret_pointer_cast<const DataTypeNullable>(return_type);
+        res = ColumnNullable::create(null_type->get_nested_type()->create_column(),
+                                     ColumnUInt8::create(size, 0));
+
+        for (int row = 0; row < size; ++row) {
+            if (column_x->is_null_at(row) || column_y->is_null_at(row)) {
+                res->insert_data(nullptr, 0);
+                continue;
+            }
+            GeoPoint point;
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/vec/functions/functions_geo.cpp:50]: same to the `buf`，just call `clear()` before use

**Comment by Gabriel39** [comment]
> and it's best to provide function UT

@zenoyang Thanks for reminding! I have added UTs for these functions and push again

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**